prometheus.php

```diff
@@ -1,37 +1,47 @@
 <?php
-
+use Automattic\VIP\Feature;
 use Automattic\VIP\Prometheus\APCu_Collector;
 use Automattic\VIP\Prometheus\Cache_Collector;
 use Automattic\VIP\Prometheus\Login_Stats_Collector;
 use Automattic\VIP\Prometheus\OpCache_Collector;
 use Automattic\VIP\Prometheus\Post_Stats_Collector;
-
+use Automattic\VIP\Prometheus\Error_Stats_Collector;
 // @codeCoverageIgnoreStart -- this file is loaded before tests start
 if ( defined( 'ABSPATH' ) ) {
+	require_once __DIR__ . '/prometheus/index.php';
+
 	$files = [
-		'/prometheus/index.php',
 		'/prometheus-collectors/class-cache-collector.php',
 		'/prometheus-collectors/class-apcu-collector.php',
 		'/prometheus-collectors/class-opcache-collector.php',
 		'/prometheus-collectors/class-login-stats-collector.php',
-		'/prometheus-collectors/class-post-stats-collector.php',
+		'/prometheus-collectors/class-error-stats-collector.php',
 	];
 
+	$should_enable_post_collector = Feature::is_enabled( 'prom-post-collection' ) || ( defined( 'VIP_GO_ENV' ) && 'production' !== VIP_GO_ENV );
+
+	if ( $should_enable_post_collector ) {
+		$files[] = '/prometheus-collectors/class-post-stats-collector.php';
+	}
+
 	foreach ( $files as $file ) {
 		if ( ! file_exists( __DIR__ . $file ) ) {
-			return; // Bail early if one of the files doesn't exist.
+			continue; // Skip to the next one
 		} else {
 			require_once __DIR__ . $file;
 		}
 	}
 
 	add_filter( 'vip_prometheus_collectors', function ( array $collectors, string $hook ): array {
 		if ( 'vip_mu_plugins_loaded' === $hook ) {
-			$collectors[] = new Cache_Collector();
-			$collectors[] = new APCu_Collector();
-			$collectors[] = new OpCache_Collector();
-			$collectors[] = new Login_Stats_Collector();
-			$collectors[] = new Post_Stats_Collector();
+			$collectors['cache']   = new Cache_Collector();
+			$collectors['apcu']    = new APCu_Collector();
+			$collectors['opcache'] = new OpCache_Collector();
+			$collectors['login']   = new Login_Stats_Collector();
+			if ( class_exists( Post_Stats_Collector::class ) ) {
+				$collectors['post'] = new Post_Stats_Collector();
+			}
+			$collectors['error'] = new Error_Stats_Collector();
```

Comment:
If we do not bail when one of the file does not exist, we have to guard every `new` with `if ( class_exists() )`.

Reply:
yeah I was conflicted about this, maybe something like

```php
$c = [
 $slug => $class
];

foreach( $c as $slug => $class ) {
  if ( class_exists() ) {
     $collectors[ $slug ] => new $class();
  }
}

Reply:
Could do more OOP and add a `addCollector` or something